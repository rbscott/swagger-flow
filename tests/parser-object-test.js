import yaml from 'js-yaml';

import parse from '../src/parser';

describe('parse', () => {
  describe('object', () => {
    it('Should parse a basic object', () => {
      const input = yaml.safeLoad(`
definitions:
  basicObject:
    type: object
    properties:
      field1:
        type: string
`);
      expect(parse(input).items).toMatchSnapshot();
    });

  });
});
