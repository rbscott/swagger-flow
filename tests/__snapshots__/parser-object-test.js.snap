// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse object Should parse a basic object 1`] = `
Array [
  Object {
    "additionalProperties": undefined,
    "fields": Array [
      Array [
        "field1",
        Object {
          "id": "#/definitions/basicObject/Field_field1",
          "name": "basicObject_Field_field1",
          "type": "string",
        },
      ],
    ],
    "id": "#/definitions/basicObject",
    "name": "basicObject",
    "required": Array [],
    "type": "object",
  },
]
`;

exports[`parse object Should parse the required field object 1`] = `
Array [
  Object {
    "additionalProperties": undefined,
    "fields": Array [
      Array [
        "field1",
        Object {
          "id": "#/definitions/basicObject/Field_field1",
          "name": "basicObject_Field_field1",
          "type": "string",
        },
      ],
    ],
    "id": "#/definitions/basicObject",
    "name": "basicObject",
    "required": Array [
      "field1",
    ],
    "type": "object",
  },
]
`;
